### PR TITLE
Add boots of blinding speed to the wizard spellbook.

### PR DESCRIPTION
**Pull request by Allspice**
:cl:
* rscadd: Boots of blinding speed added to wiz spellbook, 4x movespeed but you're blinded while you move.

### CODE REVIEW

**Comment by DamianX** [comment]
image

File: code/game/gamemodes/wizard/artefact.dm [hunk 1]
```diff
@@ -306,4 +306,31 @@
...
+	..()
+	if(istype(H) && H.get_item_by_slot(slot_shoes) == src && equipped_slot != null && equipped_slot == slot_shoes)
+		H.movement_speed_modifier *= speed_modifier
+		//H.change_sight(adding = BLIND)
+
+
+/obj/item/clothing/shoes/blindingspeed/step_action()
+	var/mob/living/carbon/human/H = loc
+	H.change_sight(adding = BLIND)
+	spawn(3)
```
Comment (DamianX) [inline code/game/gamemodes/wizard/artefact.dm]: surely there's a better way that doesn't spawn every time you take a step

Comment (Allspice) [reply code/game/gamemodes/wizard/artefact.dm]: There might be, but I'm not sure there's a check that happens when you stop moving.

**Comment by Durgen** [comment]
They cost 20 points? Seems high

File: code/game/gamemodes/wizard/artefact.dm (hunk 1 above)
Comment (DamianX) [reply code/game/gamemodes/wizard/artefact.dm]: The invisibility genetic mutation starts doing its thing when you stand still, maybe you could look at how that does it.

Comment (Allspice) [reply code/game/gamemodes/wizard/artefact.dm]: Invis relegates its stuff to Life(), which takes 0-2 seconds or so to happen. I did a little more testing, and 0-2 seconds doesn't feel unacceptably terrible compared to .5 seconds or so after you stop.

**Comment by SonixApache** [comment]
20 is the cost of one spell

File: code/game/gamemodes/wizard/artifacts.dm [hunk 2]
```diff
@@ -299,3 +299,13 @@
...
+
+/datum/spellbook_artifact/blindingspeed
+	name = "Boots of Blinding Speed"
+	abbreviation = "BS"
+	desc = "Makes you much faster, but blinds you while you move."
+	price = Sp_BASE_PRICE
+	spawned_items = list(/obj/item/clothing/shoes/blindingspeed)
+
+/datum/spellbook_artifact/blindingspeed/purchased(mob/living/carbon/human/H)
+	..()
```
Comment (Fikou) [inline code/game/gamemodes/wizard/artifacts.dm]: uhh ..() calls parent's return value, and children do that by default, so this line is completely useless

File: code/game/gamemodes/wizard/artifacts.dm [hunk 3]
```diff
@@ -299,3 +299,13 @@
 	H.add_spell(new/spell/targeted/create_trinket)
 	H.add_spell(new/spell/targeted/cool_object)
 	H.add_spell(new/spell/targeted/warm_object)
+
+/datum/spellbook_artifact/blindingspeed
+	name = "Boots of Blinding Speed"
+	abbreviation = "BS"
+	desc = "Makes you much faster, but blinds you while you move."
+	price = Sp_BASE_PRICE
```
Comment (Fikou) [inline code/game/gamemodes/wizard/artifacts.dm]: this is useless too, inherits price from parent

**Comment by spoonylove** [comment]
Will buying scrying orb nullify the blindness?

**Comment by Durgen** [comment]
@sonixapache yeah, seems high

**Comment by Probe1** [comment]
@Allspice see @Fikou 's review and then when the changes are made we're good to go.

**Comment by Killette2** [comment]
If it's cheaper such as 10 points, a wizard is more likely to sell them or buy them in the first place.

**Comment by Kurfursten** [comment]
needs 🆑

**Comment by robosperg** [comment]
boot icon sprite is reverse of most other footwear

**Comment by Allspice** [comment]
Blindness goes away in handle_regular_hud_updates every tick, same as "taking off a blindfold" or whatever.

**Comment by Kurfursten** [comment]
Seems to have a janky effect where your blindness ends sooner if you get in right before a life tick... But let's see how it goes in practice. I'm sure we'll hear about it if it feels bad.

**Pull request merged**